src/crimson/osd/pg.cc

```diff
@@ -874,38 +874,75 @@ PG::do_osd_ops_execute(
     [success_func=std::move(success_func), rollbacker, this, failure_func_ptr]
     (auto submitted_fut, auto _pre_all_completed_fut) mutable {
 
-    auto pre_all_completed_fut = _pre_all_completed_fut.safe_then_interruptible_tuple(
-      std::move(success_func),
-      crimson::ct_error::object_corrupted::handle(
-      [rollbacker, this] (const std::error_code& e) mutable {
+    auto pre_all_completed_fut =
+      _pre_all_completed_fut.safe_then_interruptible([] {
+        return OpsExecuter::osd_op_errorator::now();
+      }, OpsExecuter::osd_op_errorator::pass_further{}
+    );
+
+    /*
+     WIP: The following line (900) is not correct.
+          pre_all_completed_fut should be passed
+          as all_completed_fut (pg_rep_op_fut_t's third param).
+          The probelem with scheduling `pre_all_completed_fut`
+          here is that it'll be executed *before*
+          `submitted_fut` (pg_rep_op_fut_t's first param).
+          And won't be bounded to ClientRequest::do_process
+          use of the returned pg_rep_op_fut_t.
+
+          However, the error handling phase (which should take
+          place here) is only possible after executing `pre_all_completed_fut`.
+
+          TODO: 1) Is it possible to handle `pre_all_completed_fut`
+                   the errors before actaully getting here?
+
```

Comment:
I'm not entirely sure I'm following your intent here, but I think the problem is that splitting error_log_fut from submitted_fut doesn't help and actually makes this more complicated.  Ultimately, both error_log_fut and submitted_fut encode the same thing -- that we've fully updated the in-memory PG state, submitted (ordered) the resulting messages, and are ready to process a new operation.  Whether this was via the normal repop pathway or the error log pathway doesn't matter.  I don't think the caller cares about the difference.

I think this probably means that the submit_error_log call should be moved out of do_osd_ops into do_osd_ops_execute.  The submitted_fut should encode the normal repop path if there are no errors and the submit_error_log path if there are.

I think the failure_func argument to do_osd_ops_execute is probably useless at that point.  If the caller needs to know, the error value can be embedded into either all_completed_fut or submitted_fut (not as an error -- just as a normal value).